src/OmnichannelChatSDK.ts

```diff
@@ -446,111 +446,117 @@ class OmnichannelChatSDK {
+                    const telemetryData = {
+                        RequestId: this.requestId,
+                        ChatId: this.chatToken.chatId as string,
+                    };
 
-            try {
-                await this.AMSClient?.initialize({
-                    chatToken: this.chatToken as OmnichannelChatToken
-                });
-            } catch (error) {
-                const telemetryData = {
-                    RequestId: this.requestId,
-                    ChatId: this.chatToken.chatId as string,
-                };
+                    exceptionThrowers.throwMessagingClientConversationJoinFailure(error, this.scenarioMarker, TelemetryEvent.StartChat, telemetryData);
+                }
+            } else {
+                try {
+                    await this.IC3Client.initialize({
+                        token: this.chatToken.token,
+                        regionGtms: this.chatToken.regionGTMS,
+                        visitor: true
+                    });
+                } catch (error) {
+                    const telemetryData = {
+                        RequestId: this.requestId,
+                        ChatId: this.chatToken.chatId as string,
+                    };
 
-                exceptionThrowers.throwMessagingClientInitializationFailure(error, this.scenarioMarker, TelemetryEvent.StartChat, telemetryData);
-            }
-        } else {
-            try {
-                await this.IC3Client.initialize({
-                    token: this.chatToken.token,
-                    regionGtms: this.chatToken.regionGTMS,
-                    visitor: true
-                });
-            } catch (error) {
-                const telemetryData = {
-                    RequestId: this.requestId,
-                    ChatId: this.chatToken.chatId as string,
-                };
+                    exceptionThrowers.throwMessagingClientInitializationFailure(error, this.scenarioMarker, TelemetryEvent.StartChat, telemetryData);
+                }
 
-                exceptionThrowers.throwMessagingClientInitializationFailure(error, this.scenarioMarker, TelemetryEvent.StartChat, telemetryData);
+                try {
+                    this.conversation = await this.IC3Client.joinConversation(this.chatToken.chatId);
+                    this.scenarioMarker.completeScenario(TelemetryEvent.StartChat, {
+                        RequestId: this.requestId,
+                        ChatId: this.chatToken.chatId as string
+                    });
+                } catch (error) {
+                    const telemetryData = {
+                        RequestId: this.requestId,
+                        ChatId: this.chatToken.chatId as string,
+                    };
+
+                    exceptionThrowers.throwMessagingClientConversationJoinFailure(error, this.scenarioMarker, TelemetryEvent.StartChat, telemetryData);
+                }
             }
+        };
 
+        const amsClientPromise = async () => {
```

Comment:
I'd suggested to add a check on `this.liveChatVersion === LiveChatVersion.V2`. This client should only be initialized on V2

Reply:
`attachmentClientPromise`?